### PR TITLE
Reduce build time on all three platforms and introduce linux tests

### PR DESCRIPTION
**Pull request by RyeMutt**
* Enable precompiled header for mac and linux builds
* Improves build times on platforms by tweaking the precompiled header(6-8 minutes off mac GHA)
* Fixes various tests disabled on non-windows platforms or otherwise(llcorehttp requests, llurlentry, llkdu)
* Fix test builds on linux and enable in GHA
* Changes Linux GHA builds to use GCC for better C++20 compat

### CODE REVIEW

File: indra/newview/llinventoryitemslist.cpp [hunk 1]
```diff
@@ -99,7 +99,7 @@ void LLInventoryItemsList::updateSelection()
 
     for(std::vector<LLSD>::const_iterator cur_id_it = cur.begin(); cur_id_it != cur.end() && !mSelectTheseIDs.empty(); ++cur_id_it)
     {
-        uuid_vec_t::iterator select_ids_it = std::find(mSelectTheseIDs.begin(), mSelectTheseIDs.end(), *cur_id_it);
+        uuid_vec_t::iterator select_ids_it = std::find(mSelectTheseIDs.begin(), mSelectTheseIDs.end(), cur_id_it->asUUID());
```
Comment (Geenz) [inline indra/newview/llinventoryitemslist.cpp:102]: Noting from conversation with Rye: this is due to the PCH stuff in this PR.  This makes the type resolution more clear.

File: indra/llcorehttp/examples/http_texture_load.cpp
Comment (Geenz) [inline indra/llcorehttp/examples/http_texture_load.cpp:1]: Speaking with Rye: The removals here are due to dead OpenSSL inits that we don't really use.

**Comment by Geenz** [comment]
Some benchmarks on my machine (Ryzen 9 7945HX3D - 16C/32T) using Release configuration:
develop: 08:21.696 minutes
rye/buildtests without LTO: 06:33.442 minutes
rye/buildtests with LTO (`autobuild configure -- -DUSE_LTO=ON`): 05:49.102 minutes

**Comment by marchcat** [comment]
Ryzen 9 5950X (16C/32T), Release configuration:

develop: 07:46.027 minutes
rye/buildtests without LTO: 05:40.638 minutes, 27% faster than develop 🚀
rye/buildtests with LTO (autobuild configure -- -DUSE_LTO=ON): 06:04.335 minutes, 22% faster than develop

Enabling LTO gives me `Previous IPDB not found, fall back to full compilation.` in the log, so it takes longer, but it's still a noticeable improvement.

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -264,13 +267,6 @@ jobs:
           fi
           export PYTHON_COMMAND_NATIVE="$(native_path "$PYTHON_COMMAND")"
 
-          # Compile with clang, link with mold on linux.
-          if [[ "$RUNNER_OS" == "Linux" ]]; then
-            export CC=clang
-            export CXX=clang++
-            export CMAKE_OPTIONS='-DLINK_WITH_MOLD=ON'
-          fi
```
Comment (brad-linden) [inline .github/workflows/build.yaml:272]: do we actually want to switch off of clang?  I thought we were trying to work towards clang on all 3 platforms?

**Review by brad-linden — APPROVED** [review]

**Pull request merged**